src/vs/workbench/api/common/extHostDocumentSaveParticipant.ts

```diff
@@ -143,13 +152,13 @@ export class ExtHostDocumentSaveParticipant implements ExtHostDocumentSavePartic
 		}).then(values => {
 			const dto: IWorkspaceEditDto = { edits: [] };
 			for (const value of values) {
-				if (Array.isArray(value) && (<vscode.TextEdit[]>value).every(e => e instanceof TextEdit)) {
+				if (Array.isArray(value) && value.every(e => isITextEdit(e))) {
```

Comment:
This is technically correct. But instead of relaxing the API check here, which might have more unintented consequences, let's simply return what the API expects, namely `vscode.TextEdit` objects.

Reply:
The reason I changed this is because for the following value of `e`:

<img width="717" alt="Screenshot 2023-07-28 at 14 46 40" src="https://github.com/microsoft/vscode/assets/61460952/b4e849c0-7acd-46ea-879d-e73c9fcd7d80">

It appears it is not correctly cast to the type `TextEdit` with the initial code (despite containing the fields `newText` and `range`. Hence the check `e instanceof TextEdit` returns false.